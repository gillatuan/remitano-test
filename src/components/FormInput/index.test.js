/* eslint-disable testing-library/prefer-screen-queries */
import { FormInput } from "./index"
import { render, waitFor } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { testSnapshots } from 'test'
import { Form } from 'react-bootstrap'

const makeSut = (props) => {
  const { ariaLabel, error, id, isView, label, name, placeholder, value } = props
  return render(
    <FormInput
      ariaLabel={ariaLabel}
      error={error}
      id={id}
      isView={isView}
      label={label}
      name={name}
      placeholder={placeholder}
      required={true}
      value={value}
      onChange={jest.fn()}
    />
  )
}

const initialConfig = {
  ariaLabel: "aria-input-title",
  label: "Form Title",
  id: "input-title",
  name: "title",
}

describe('<InputField />', () => {
  testSnapshots(Form.Control, [
    {
      props: initialConfig,
      description: 'renders input correctly'
    },
    {
      props: {
        ...initialConfig,
        disabled: true,
      },
      description: 'renders disable input correctly'
    },
    {
      props: {
        ...initialConfig,
        readOnly: true,
      },
      description: 'renders readonly input'
    },
    {
      props: {
        ...initialConfig,
        type: 'email'
      },
      description: 'renders input is email'
    },
    {
      props: {
        ...initialConfig,
        type: 'select'
      },
      description: 'renders input is select'
    },
    {
      props: {
        ...initialConfig,
        multiple: true,
        type: 'select'
      },
      description: 'renders input is select multi'
    },
    {
      props: {
        ...initialConfig,
        type: 'file'
      },
      description: 'renders input is file'
    },
    {
      props: {
        ...initialConfig,
        type: 'textarea'
      },
      description: 'renders input is textarea'
    },
    {
      props: {
        ...initialConfig,
        type: 'search'
      },
      description: 'renders input is search'
    },
    {
      props: {
        ...initialConfig,
        type: 'image'
      },
      description: 'renders input is image'
    },
    {
      props: {
        ...initialConfig,
        type: 'number'
      },
      description: 'renders input is number'
    },
    {
      props: {
        ...initialConfig,
        type: 'date'
      },
      description: 'renders input is date'
    },
    {
      props: {
        ...initialConfig,
        type: 'checkbox'
      },
      description: 'renders input is checkbox'
    },
    {
      props: {
        ...initialConfig,
        disabled: true,
        type: 'checkbox'
      },
      description: 'renders input is disabled checkbox'
    },
    {
      props: {
        ...initialConfig,
        type: 'radio'
      },
      description: 'renders input is radio'
    },
    {
      props: {
        ...initialConfig,
        disabled: true,
        type: 'radio'
      },
      description: 'renders input is disabled radio'
    },
  ])
})

describe("<FormInput />", () => {
  test("Should render label correctly", () => {
    const { getByText } = makeSut(initialConfig)

    // eslint-disable-next-line testing-library/prefer-screen-queries
    expect(getByText(/Form Title/)).toBeInTheDocument()
  })

  test("Should call onChange value successfully", async () => {
    const { getByTestId } = makeSut(initialConfig)
    userEvent.type(getByTestId("input-title"), "abc")
    await waitFor(() => expect(getByTestId("input-title")).toHaveDisplayValue("abc"))
  })
})
