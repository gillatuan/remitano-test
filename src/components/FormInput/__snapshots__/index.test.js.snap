// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> snapshots renders disable input correctly 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  disabled={true}
  id="input-title"
  label="Form Title"
  name="title"
/>
`;

exports[`<InputField /> snapshots renders input correctly 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
/>
`;

exports[`<InputField /> snapshots renders input is checkbox 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="checkbox"
/>
`;

exports[`<InputField /> snapshots renders input is date 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="date"
/>
`;

exports[`<InputField /> snapshots renders input is disabled checkbox 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  disabled={true}
  id="input-title"
  label="Form Title"
  name="title"
  type="checkbox"
/>
`;

exports[`<InputField /> snapshots renders input is disabled radio 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  disabled={true}
  id="input-title"
  label="Form Title"
  name="title"
  type="radio"
/>
`;

exports[`<InputField /> snapshots renders input is email 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="email"
/>
`;

exports[`<InputField /> snapshots renders input is file 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="file"
/>
`;

exports[`<InputField /> snapshots renders input is image 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="image"
/>
`;

exports[`<InputField /> snapshots renders input is number 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="number"
/>
`;

exports[`<InputField /> snapshots renders input is radio 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="radio"
/>
`;

exports[`<InputField /> snapshots renders input is search 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="search"
/>
`;

exports[`<InputField /> snapshots renders input is select 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="select"
/>
`;

exports[`<InputField /> snapshots renders input is select multi 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  multiple={true}
  name="title"
  type="select"
/>
`;

exports[`<InputField /> snapshots renders input is textarea 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  type="textarea"
/>
`;

exports[`<InputField /> snapshots renders readonly input 1`] = `
<input
  ariaLabel="aria-input-title"
  className="form-control"
  id="input-title"
  label="Form Title"
  name="title"
  readOnly={true}
/>
`;
